src/utils/helpers.js

```diff
@@ -7,12 +7,44 @@ function signPoly(value) {
 }
 export const sign = Math.sign || signPoly
 
+/**
+ * Checks if the flag is set
+ * @param val
+ * @param flag
+ * @returns {boolean}
+ */
+function hasFlag(val, flag) {
+    return (val & flag) === flag
```

Comment:
xD, once serviceparadice said that he don't give a damn about speed/performance. Also it's should be `==`, not `===`, there's no ways to get diff type than integer.

Reply:
The fact that the type wont be different is not a really good reason to use `==` instead of `===`. Do you have any source to support this? For me, the only good reason to use `==` is if you absolutely need type coercion. In every other cases, we should prefer strict equality `===`. Talking about performance here is also weird since the difference between the two will always be negligible and `===` will be faster in most cases.

Here are some references about this:
https://codeahoy.com/javascript/2019/10/12/==-vs-===-in-javascript/
https://stackoverflow.com/a/359509/2774496
https://stackoverflow.com/a/8045526/2774496

Reply:
Do you know bit operations return type? (Yes, just check typeof) It's everytime int, you don't really need === there, but whatever. Instead providing js links, which are describe js, check c bit operations which are implemented inside js engines there's no way to get anything else than integer, js doesn't work with 64, 128 bits numbers on bitwise operations.

Reply:
I know what a bitwise operation is and I know its return type. There are still no reasons to not use strict equality.

Reply:
What the reason use strict equality if there's no way to get anything instead number?)
let a = 16;
b = 2;
alert(typeof (a&b));

Idk why u sent these links.

Reply:
The reason is it's safer coding for a lot of other operations so linters and code formatting tools will always force/highly recommend you to use them. Btw you can't assume the type of flag, it should be a number and an actual flag but it's not type checked so if you passed `true` in there well you'd be surprised by what you'd get
```
(1 & true) === true
false
(1 & true) == true
true
```